### PR TITLE
Iox 1960 update integrationtests

### LINKED ISSUE #1906: integrationtest CI job failing
## Required information

**Eclipse iceoryx version:**
`master` branch

**Observed result or behaviour:**
The CI job for executing is failing very often: https://github.com/eclipse-iceoryx/iceoryx/actions/runs/4195696582/jobs/7290325245

We may need to update the ROS distribution of the integrationtests to a newer LTS distribution because ROS Foxy will be deprecated in May 2023.

**Expected result or behaviour:**
- CI job is stable and green
- Integrationtests updated to ROS Humble

**Conditions where it occurred / Performed steps:**
Describe how one can reproduce the bug.

### PR DESCRIPTION
**Pull request by dkroenke**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] All touched (C/C++) source code files from `iceoryx_hoofs` are added to `./clang-tidy-diff-scans.txt`
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #1906

### CODE REVIEW

File: iceoryx_integrationtest/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2021 - 2022 by Apex.AI Inc. All rights reserved.
+# Copyright (c) 2021, 2023 by Apex.AI Inc. All rights reserved.
```
Comment (mossmaurice) [inline iceoryx_integrationtest/CMakeLists.txt:1]: Nitpick: 
```suggestion
# Copyright (c) 2021 - 2023 by Apex.AI Inc. All rights reserved.
```

File: iceoryx_integrationtest/CMakeLists.txt [hunk 2]
```diff
@@ -20,29 +20,22 @@ project(iceoryx_integrationtest)
...
-  add_ros_test(iceoryx_integrationtest/test_request_response_listener.py)
-  add_ros_test(iceoryx_integrationtest/test_request_response_untyped.py)
-  add_ros_test(iceoryx_integrationtest/test_icediscovery_in_c_example.py)
+  list(APPEND IOX_INT_TESTS test_roudi_startup_shutdown test_callback_example test_callback_in_c_example)
+  list(APPEND IOX_INT_TESTS test_icehello_example test_icedelivery_example test_icedelivery_in_c_example)
+  list(APPEND IOX_INT_TESTS test_singleprocess_example test_waitset_example test_waitset_in_c_example)
+  list(APPEND IOX_INT_TESTS test_iceoptions_example test_complexdata_example test_user_header_example)
+  list(APPEND IOX_INT_TESTS test_icediscovery_example test_icediscovery_in_c_example test_icediscovery_monitoring)
+  list(APPEND IOX_INT_TESTS test_request_response_in_c_listener_and_waitset test_request_response_basic test_request_response_in_c_basic)
+  list(APPEND IOX_INT_TESTS test_request_response_listener test_request_response_untyped)
```
Comment (mossmaurice) [inline iceoryx_integrationtest/CMakeLists.txt:31]: @dkroenke Why not just one `list(APPEND ..)` call?

File: iceoryx_integrationtest/package.xml [hunk 3]
```diff
@@ -21,6 +21,22 @@
...
+  <test_depend>example_icediscovery</test_depend>
+  <test_depend>example_icediscovery_in_c</test_depend>
+  <test_depend>example_icehello</test_depend>
+  <test_depend>example_iceoptions</test_depend>
+  <test_depend>example_request_response</test_depend>
+  <test_depend>example_request_response_in_c</test_depend>
+  <test_depend>example_singleprocess</test_depend>
+  <test_depend>example_user_header</test_depend>
+  <test_depend>example_waitset</test_depend>
+  <test_depend>example_waitset_in_c</test_depend>
```
Comment (mossmaurice) [inline iceoryx_integrationtest/package.xml:38]: Maybe add the dependency could be a sub-point in [this list](https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#how-to-add-a-new-example)?

**Review by mossmaurice — DISMISSED** [review]
LGTM, I left a few nitpicks.

File: iceoryx_integrationtest/package.xml [hunk 4]
```diff
@@ -21,6 +21,22 @@
...
+
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_integrationtest/package.xml:39]: What about `example_ice_access_control`?

Comment (dkroenke) [reply iceoryx_integrationtest/package.xml:39]: @FerdinandSpitzschnueffler The `example_ice_access_control` is not tested via the iceoryx_integrationtests

File: iceoryx_integrationtest/CMakeLists.txt (hunk 1 above)
Comment (dkroenke) [reply iceoryx_integrationtest/CMakeLists.txt]: @mossmaurice Fixed that

File: iceoryx_integrationtest/package.xml (hunk 3 above)
Comment (dkroenke) [reply iceoryx_integrationtest/package.xml:38]: I added some extra words

File: iceoryx_integrationtest/package.xml (hunk 4 above)
Comment (dkroenke) [reply iceoryx_integrationtest/package.xml:40]: @FerdinandSpitzschnueffler Right, the example_ice_access_control was missing in the list and tests, thanks for the hint!

**Review by FerdinandSpitzschnueffler — DISMISSED** [review]

**Review by mossmaurice — APPROVED** [review]

File: iceoryx_integrationtest/package.xml (hunk 4 above)
Comment (dkroenke) [reply iceoryx_integrationtest/package.xml:40]: @mossmaurice @FerdinandSpitzschnueffler It seems that was somehow never used as integrationtest and the test code there needs more effort in testing and refactoring, let's address tis in issue #1910.

**Review by mossmaurice — DISMISSED** [review]

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

**Review by elfenpiff — APPROVED** [review]

**Pull request merged**